components/webhook-mgt/org.wso2.carbon.identity.webhook.management/src/main/java/org/wso2/carbon/identity/webhook/management/api/service/EventSubscriberService.java

```diff
@@ -0,0 +1,105 @@
+     *
+     * @param webhook      The webhook to subscribe.
+     * @param tenantDomain Tenant domain.
+     * @return True if subscription is successful, false otherwise.
+     * @throws WebhookMgtException If an error occurs during subscription.
+     */
+    public boolean subscribe(Webhook webhook, String tenantDomain) throws WebhookMgtException {
+
+        List<EventSubscriber> subscribers =
+                WebhookManagementComponentServiceHolder.getInstance().getEventSubscribers();
+        if (subscribers.isEmpty()) {
+            LOG.warn("No webhook subscribers registered. Webhook subscription will not be processed " +
+                    "by external systems.");
+            return false;
+        }
+
+        for (EventSubscriber subscriber : subscribers) {
+            try {
+                boolean success =
+                        subscriber.subscribe(webhook.getEventsSubscribed(), webhook.getEndpoint(), tenantDomain);
+                if (!success) {
+                    LOG.error(
+                            "Subscriber " + subscriber.getName() + " failed to subscribe webhook: " + webhook.getId());
+                }
+            } catch (WebhookMgtException e) {
+                LOG.error("Error occurred while subscribing webhook: " + webhook.getId() +
+                        " with subscriber: " + subscriber.getName(), e);
+            }
+        }
+        return true;
+    }
+
+    /**
+     * Unsubscribe an event from external systems.
+     * This method delegates to appropriate event subscribers.
+     *
+     * @param webhook      The webhook to unsubscribe.
+     * @param tenantDomain Tenant domain.
+     * @return True if unsubscription is successful, false otherwise.
+     * @throws WebhookMgtException If an error occurs during unsubscription.
+     */
+    public boolean unsubscribe(Webhook webhook, String tenantDomain) throws WebhookMgtException {
+
+        List<EventSubscriber> subscribers =
+                WebhookManagementComponentServiceHolder.getInstance().getEventSubscribers();
+        if (subscribers.isEmpty()) {
+            LOG.warn("No webhook subscribers registered. Webhook unsubscription will not be processed " +
+                    "by external systems.");
+            return false;
+        }
+        for (EventSubscriber subscriber : subscribers) {
+            try {
+                boolean success =
+                        subscriber.unsubscribe(webhook.getEventsSubscribed(), webhook.getEndpoint(), tenantDomain);
+                if (!success) {
+                    LOG.warn("Subscriber " + subscriber.getName() + " failed to unsubscribe webhook: " +
+                            webhook.getId());
+                }
+            } catch (WebhookMgtException e) {
+                LOG.error("Error occurred while unsubscribing webhook: " + webhook.getId() +
+                        " from subscriber: " + subscriber.getName(), e);
+            }
```

Comment:
same here. exception should be thrown

Reply:
Handle separately https://github.com/wso2/carbon-identity-framework/pull/6741#discussion_r2098149320